src/Microsoft.Performance.SDK.Runtime/ProcessingSourceExecutor.cs

```diff
@@ -155,5 +186,45 @@ public async Task<ExecutionResult> ExecuteAsync(
                 this.failedToEnableTables,
                 metadataName);
         }
+
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
+            {
+                if (disposing)
+                {
+                    if ((this.Processor is not null) && (this.Processor is IDisposable disposable))
+                    {
+                        this.Context?.ProcessingSource?.Instance?.DisposeProcessor(this.Processor);
```

Comment:
Should we still call this even if the processor is not an `IDisposable`?

Reply:
Hrm, that's a good question. I'll go ahead and do that.